Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/VisualUtils.cs

```diff
@@ -211,6 +211,37 @@ static internal Vector3 GetLinkPosition(int linkIndex, ref Vector3[] cornerPoint
             return Vector3.zero;
         }
 
+        /// <summary>
+        /// Util function for retrieving a position for the given face index of a box.
+        /// This method makes sure all visual components are having the same definition of face centers.
+        /// </summary>
+        /// <param name="faceIndex">Index of the face the position is queried for.</param>
+        /// <param name="cornerPoints">Corner points array of the box.</param>
+        /// <returns>Center position of face.</returns>
+        static internal Vector3 GetFaceCenterPosition(int faceIndex, ref Vector3[] cornerPoints)
+        {
+            Debug.Assert(cornerPoints != null && cornerPoints.Length == 8, "Invalid corner points array passed");
+            if (cornerPoints != null && cornerPoints.Length == 8)
+            {
+                switch (faceIndex)
+                {
+                    case 0:
```

Comment:
From a readability side, it might be helpful to have some other define that's like:

enum Face {
    ForwardX = 0,
ForwardX = 0,
ForwardX = 0,
ForwardX = 0,
Face

Reply:
@thalbern if we implement this change, this should apply for all of the other handles too. The other handles use anonymous indices for keeping track of handles too. Would appreciate your thoughts on this.

Reply:
added enum for edges and faces